### PR TITLE
upgrade to engine_cart 0.8.0

### PR DESCRIPTION
**Pull request by ndushay**
@cbeer @peetucket 

(I got here by mistake, but I did it anyway)

### CODE REVIEW

**Comment by ndushay** [comment]
@cbeer   the tests were running ok locally, but I am also getting this failure on spotlight-dor-resources, which was updated to engine_cart 0.8.0 recently ...

Seemingly, the relevant stack track is this:
```
LoadError: cannot load such file -- spotlight-dor-resources
/Users/ndushay/.rvm/gems/ruby-2.2.2/gems/bundler-1.10.6/lib/bundler/runtime.rb:76:in `require'
/Users/ndushay/.rvm/gems/ruby-2.2.2/gems/bundler-1.10.6/lib/bundler/runtime.rb:76:in `block (2 levels) in require'
/Users/ndushay/.rvm/gems/ruby-2.2.2/gems/bundler-1.10.6/lib/bundler/runtime.rb:72:in `each'
/Users/ndushay/.rvm/gems/ruby-2.2.2/gems/bundler-1.10.6/lib/bundler/runtime.rb:72:in `block in require'
/Users/ndushay/.rvm/gems/ruby-2.2.2/gems/bundler-1.10.6/lib/bundler/runtime.rb:61:in `each'
/Users/ndushay/.rvm/gems/ruby-2.2.2/gems/bundler-1.10.6/lib/bundler/runtime.rb:61:in `require'
/Users/ndushay/.rvm/gems/ruby-2.2.2/gems/bundler-1.10.6/lib/bundler.rb:134:in `require'
/Users/ndushay/spotlight/spotlight-dor-resources/.internal_test_app/config/application.rb:7:in `<top (required)>'
/Users/ndushay/spotlight/spotlight-dor-resources/.internal_test_app/Rakefile:4:in `require'
/Users/ndushay/spotlight/spotlight-dor-resources/.internal_test_app/Rakefile:4:in `<top (required)>'
```

**Comment by ndushay** [comment]
@jkeck  wanna merge this?  Peter and @cbeer  aren't here today.   

Or, if you'd rather, merge Chri's fix for projectblacklight/blacklight#1306, cut a new patch release for blacklight, and I'll revise this and then ask you to merge.   (The pinning will go away soon in either case ... but I need to get it addressed so I can have tests working for spotlight-dor-resources so I can work on full text with hit highlighting so I can live in the house that Jill built.)

File: .rubocop.yml [hunk 1]
```diff
@@ -13,9 +14,12 @@ AllCops:
 # Metrics/LineLength:
 #   Max: 120
 
+
 Style/StringLiterals:
   Enabled: true
   EnforcedStyle: single_quotes
+  Exclude:
+    - 'Gemfile'  # until engine_cart stanza is violation free
```
Comment (jkeck) [inline .rubocop.yml]: Is this just as a reminder to update the Gemfile later? It seems like we could make the Gemfile modifications w/o causing any violations (perhaps just quote issues?)

**Comment by jkeck** [comment]
Yeah....all that sounds like a lot of work.  I just merged his PR to the 5x branch.  I **think** I know how to release Blacklight...  Up to you, I can kick off that release.  This *seems* fine I guess.

File: .rubocop.yml (hunk 1 above)
Comment (ndushay) [inline .rubocop.yml]: currently, engine_cart creates a "stanza" of chars that it puts at the end of your Gemfile.  That stanza has rubocop errors per default rubocop settings.  If you correct those rubocop errors in your Gemfile, then the stanza doesn't perfectly match any more.  In theory, engine_cart doesn't care, but I have a PR in to engine_cart  to fix the stanza.

**Comment by ndushay** [comment]
It's totally fine w me if you want to release blacklight 5.16.whatever.  It's gotta be addressed here sometime or other.  

As you can tell, this whole thing is kinda full of stupid little stuff due to upstream changes to engine_cart and to blacklight.  The fun will be coming your way sometime soon!

**Comment by cbeer** [comment]
I've added commits in this PR to make it compatible with Blacklight 5.16 (actually, require Blacklight 5.16, as I fixed the new deprecation warnings too).

I also tried to address @jkeck's concern about the Gemfile's engine_cart_stanza, as it's fair game to update that stanza to use single quotes within the Gemfile.

File: .rubocop.yml [hunk 2]
```diff
@@ -13,6 +14,7 @@ AllCops:
 # Metrics/LineLength:
 #   Max: 120
 
+
 Style/StringLiterals:
   Enabled: true
   EnforcedStyle: single_quotes
```
Comment (ndushay) [inline .rubocop.yml]: I'm ok with this, but again:  did the team agree on a style??

File: blacklight-spotlight.gemspec [hunk 3]
```diff
@@ -56,7 +56,7 @@ these collections.)
   s.add_development_dependency 'rspec-activemodel-mocks'
   s.add_development_dependency 'rspec-collection_matchers'
   s.add_development_dependency 'capybara', '>= 2.5.0'
-  s.add_development_dependency 'rubocop'
+  s.add_development_dependency 'rubocop', '~> 0.35'
```
Comment (ndushay) [inline blacklight-spotlight.gemspec]: add comment # for newer ABC metric   ?

File: lib/spotlight/engine.rb [hunk 4]
```diff
@@ -125,11 +125,13 @@ def self.blacklight_config
...
-    # Field containing the last modified date for a Solr document
-    Blacklight::Configuration.default_values[:index].timestamp_field ||= 'timestamp'
+    initializer 'blacklight.configuration' do
+      # Field containing the last modified date for a Solr document
+      Blacklight::Configuration.default_values[:index].timestamp_field ||= 'timestamp'
 
-    # Field containing the last modified date for a Solr document
-    Blacklight::Configuration.default_values[:browse] ||= Blacklight::OpenStructWithHashAccess.new(document_actions: [])
+      # Field containing the last modified date for a Solr document
+      Blacklight::Configuration.default_values[:browse] ||= Blacklight::OpenStructWithHashAccess.new(document_actions: [])
```
Comment (ndushay) [inline lib/spotlight/engine.rb]: the comment seems wrong.

**Comment by ndushay** [comment]
3 line comments;  once they are addressed, not sure who should merge as chris and I are both sort of merging our own changes.

File: .rubocop.yml (hunk 2 above)
Comment (cbeer) [inline .rubocop.yml]: I believe this is consistent with current practice in other projects.

I should remove that extra blank line, though.

Comment (ndushay) [inline .rubocop.yml]: we don't have a consistent practice in DLSS, and we don't have an agreed upon practice in this spotlight sprint series yet, either.   If it's about shutting up hound, then add a comment to that effect.

Comment (cbeer) [inline .rubocop.yml]: I'm confused. This wasn't changed as part of this commit (except for the errant whitespace, which I've removed).

File: blacklight-spotlight.gemspec (hunk 3 above)
Comment (cbeer) [inline blacklight-spotlight.gemspec]: Let me rephrase the commit message that changed it. The `AbcSize` is really incidental, it's really about making sure the team and travis are in sync with these kinds of upstream changes (or, I was annoyed that travis was failing a build that worked locally).

**Comment by ndushay** [comment]
@cbeer apparently *I* was confused on the quote style thing in .rubocop.yml.   Sorry about that.  Just not looking to add more constraints of that nature before we have a conversation.  

Wanna add a comment as to the version of rucobop in the gemspec?  After that, I'm good with this.

**Comment by cbeer** [comment]
Is the new commit message sufficient? I'm not sure there's anything to say; it's essentially an arbitrary version to get consistency across environments.

**Comment by ndushay** [comment]
Ah, I understand the rubocop version restriction better now.   Should you merge, or me, or wait for jessie or peter to do it on Monday?

**Comment by cbeer** [comment]
Great. Thanks for the comments. I think if we're both happy with it, let's ship it.

Thanks too for struggling through the `engine_cart` upgrade. It's particularly annoying that the `source_root` for the test app generator is relative to the application, and I hope to figure out a better approach some day..

**Comment by ndushay** [comment]
you gonna cut a new gem release or would you like me to do it?
